tests/Unit/TestHelpers.hpp

```diff
@@ -91,6 +91,8 @@ T serialize_and_deserialize(const T& t) {
 }
 
 /// Test for copy semantics assuming operator== is implement correctly
+#pragma GCC diagnostic push
+#pragma GCC diagnostic ignored "-Wparentheses"
```

Comment:
huh?  the warning is that it suggests extra parentheses which are not needed and break catch's useful error messages